File: framework/src/main/scala/com/software/dm/swallow/stormy/scala/security/addrlib/bdia/bean/T_url_rule.scala
package com.software.dm.swallow.stormy.scala
package security.addrlib.bdia.bean

import scala.collection.mutable.ListBuffer


class T_url_rule {
  val ACTION_BASIC_APP = 0

  private var n_id = 0L
  private var domain_one: String = null
  private var brand: String = null
  private var domain_comp: String = null
  private var service_name: String = null
  private var rule: String = null
  private var rule_type = 0
  private var rule_type_level = 0
  private var is_app_url = 0
  private var bqz_id: String = null
  private var bqz_name: String = null

  private var basicTypes: ListBuffer[T_basic_type_rel] = null

  def this(n_id: Long, domain_one: String, brand: String, domain_comp: String, service_name: String, rule: String, rule_type: Int, rule_type_level: Int, is_app_url: Int, bqz_id: String, bqz_name: String, basicTypes: ListBuffer[T_basic_type_rel]) {
    this()
    this.n_id = n_id
    this.domain_one = domain_one
    this.brand = brand
    this.domain_comp = domain_comp
    this.service_name = service_name
    this.rule = rule
    this.rule_type = rule_type
    this.rule_type_level = rule_type_level
    this.is_app_url = is_app_url
    this.bqz_id = bqz_id
    this.bqz_name = bqz_name
    this.basicTypes = basicTypes
  }

  def getN_id: Long = n_id

  def getDomain_one: String = domain_one

  def getBrand: String = brand

  def getDomain_comp: String = domain_comp

  def getService_name: String = service_name

  def getRule: String = rule

  def getRule_type: Int = rule_type

  def getRule_type_level: Int = rule_type_level

  def getIs_app_url: Int = is_app_url

  def getBqz_id: String = bqz_id

  def getBqz_name: String = bqz_name

  def getBasicTypes: ListBuffer[T_basic_type_rel] = basicTypes
}

File: framework/src/main/java/com/software/dm/swallow/stormy/security/match/bdia/bean/T_url_rule.java
package com.software.dm.swallow.stormy.security.match.bdia.bean;

import java.io.Serializable;
import java.util.List;

public final class T_url_rule implements Serializable {

    public static final int ACTION_BASIC_APP = 0;

    private long n_id;
    private String domain_one;
    private String brand;
    private String domain_comp;
    private String service_name;
    private String rule;
    private int rule_type;
    private int rule_type_level;
    private int is_app_url;
    private String bqz_id;
    private String bqz_name;

    private List<T_basic_type_rel> basicTypes;

    public T_url_rule(long n_id, String domain_one, String brand, String domain_comp, String service_name, String rule, int rule_type, int rule_type_level, int is_app_url, String bqz_id, String bqz_name, List<T_basic_type_rel> basicTypes) {

        this.n_id = n_id;
        this.domain_one = domain_one;
        this.brand = brand;
        this.domain_comp = domain_comp;
        this.service_name = service_name;
        this.rule = rule;
        this.rule_type = rule_type;
        this.rule_type_level = rule_type_level;
        this.is_app_url = is_app_url;
        this.bqz_id = bqz_id;
        this.bqz_name = bqz_name;
        this.basicTypes = basicTypes;
    }

    public long getN_id() {
        return n_id;
    }

    public String getDomain_one() {
        return domain_one;
    }

    public String getBrand() {
        return brand;
    }

    public String getDomain_comp() {
        return domain_comp;
    }

    public String getService_name() {
        return service_name;
    }

    public String getRule() {
        return rule;
    }

    public int getRule_type() {
        return rule_type;
    }

    public int getRule_type_level() {
        return rule_type_level;
    }

    public int getIs_app_url() {
        return is_app_url;
    }

    public String getBqz_id() {
        return bqz_id;
    }

    public String getBqz_name() {
        return bqz_name;
    }

    public List<T_basic_type_rel> getBasicTypes() {
        return basicTypes;
    }

}
